xonsh/commands_cache.py

```diff
@@ -73,7 +73,7 @@ def get_possible_names(name):
     @property
     def all_commands(self):
         paths = builtins.__xonsh_env__.get('PATH', [])
-        pathset = frozenset(x for x in paths if os.path.isdir(x))
+        pathset = tuple(x for x in paths if os.path.isdir(x))
```

Comment:
This should probably be a set, so that it removes duplicated entries

Reply:
Or maybe a tuple is a better option, but you need to create the tuple after the the set

Reply:
But `frozenset`s can't be reversed. Maybe should just remove the duplications old fashioned way.

Reply:
The problem is that frozensets don't preserve order. Duplications could be removed in EnvPath. Are there any environment path variables which might cause trouble if we remove duplications.